src/renderer/fonts/caching_shaper.rs

```diff
@@ -68,26 +56,30 @@ impl CachingShaper {
     }
 
     pub fn current_size(&self) -> f32 {
-        self.options
-            .as_ref()
-            .map(|options| options.size)
-            .unwrap_or(DEFAULT_FONT_SIZE)
-            * self.scale_factor
+        self.options.size * self.scale_factor
+    }
+
+    pub fn update_scale_factor(&mut self, device_scale_factor: f32) {
+        println!("Update scale_factor: {}", device_scale_factor);
+        self.scale_factor = device_scale_factor * points_to_pixels(device_scale_factor);
+        self.reset_font_loader();
     }
 
     pub fn update_font(&mut self, guifont_setting: &str) -> bool {
-        let new_options = FontOptions::parse(guifont_setting, DEFAULT_FONT_SIZE);
+        println!("Updating font: {}", guifont_setting);
```

Comment:
use a trace here rather than println

Reply:
That's temporary code until this is draft. I'll remove all prints after this is working as expected.